runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -933,13 +905,13 @@ private void process(
     final Windmill.WorkItem workItem = work.getWorkItem();
     final String computationId = computationState.getComputationId();
     final ByteString key = workItem.getKey();
-    work.setState(Work.State.PROCESSING);
+    work.setState(State.PROCESSING);
     {
-      StringBuilder workIdBuilder = new StringBuilder(33);
-      workIdBuilder.append(Long.toHexString(workItem.getShardingKey()));
-      workIdBuilder.append('-');
-      workIdBuilder.append(Long.toHexString(workItem.getWorkToken()));
-      DataflowWorkerLoggingMDC.setWorkId(workIdBuilder.toString());
+      String workIdBuilder =
```

Comment:
no longer a builder. Does Java optimize repeated string additions better now? IIRC we maybe saw cpu or objects on a profile

Reply:
I believe StringBuilder is very useful when concatenating in a loop, for simple concatenation the compiler will optimize for us. https://dzone.com/articles/string-concatenation-performacne-improvement-in-ja